src/btreemap.rs

```diff
@@ -3029,4 +3039,28 @@ mod test {
         let btree: BTreeMap<T, T, _> = BTreeMap::init(btree.into_memory());
         assert_eq!(btree.get(&T), Some(T));
     }
+
+    #[test]
+    fn test_clear_new() {
```

Comment:
For extra sanity, it would be useful to also have a version of this test where we declare a `BTreeMap` with an unbounded type.